data/src/config/buffer.rs

```diff
@@ -73,7 +77,36 @@ impl Default for ServerMessage {
             enabled: true,
             smart: Default::default(),
             username_format: UsernameFormat::default(),
+            exclude: Default::default(),
+            include: Default::default(),
+        }
+    }
+}
+
+impl ServerMessage {
+    pub fn should_send_message(&self, channel: Option<&str>) -> bool {
+        // Server Message is not enabled.
+        if !self.enabled {
+            return false;
         }
+
+        let is_channel_filtered = |list: &Vec<String>, channel: Option<&str>| -> bool {
+            let wildcards = ["*", "all"];
+            channel.map_or(false, |channel| {
+                list.iter()
+                    .any(|item| wildcards.contains(&item.as_str()) || item == channel)
+            })
+        };
+
+        let channel_included = is_channel_filtered(&self.include, channel);
+        let channel_excluded = is_channel_filtered(&self.exclude, channel);
+
+        // If the channel is included, it has precedence over excluded.
+        if channel_included || !channel_excluded {
+            return true;
+        }
+
+        false
```

Comment:
Nit: Return `channel_included || !channel_excluded` directly as it resolving to `true / false` is returning `true / false`

Reply:
Doh, of cause. Thanks.